### PR TITLE
Governorship conversion part I

### PR DESCRIPTION
**Pull request by IhateTrains**
#90

### CODE REVIEW

File: ImperatorToCK3/CK3/World.cs [hunk 1]
```diff
@@ -314,40 +363,95 @@ Imperator.World impWorld
...
+				}
+
+				if (!Provinces.ContainsKey(title.CapitalBaronyProvince)) {
+					Logger.Warn($"Capital barony province not found {title.CapitalBaronyProvince}");
+					continue;
+				}
+				var ck3CapitalBaronyProvince = Provinces[title.CapitalBaronyProvince];
+				var impProvince = ck3CapitalBaronyProvince.ImperatorProvince;
+
+				if (impProvince is not null) {
```
Comment (Idhrendur) [inline ImperatorToCK3/CK3/World.cs:384]: Reverse the condition and make the null case continue the loop. Then you can move the rest out one indentation level.

File: ImperatorToCK3/CK3/World.cs [hunk 2]
```diff
@@ -314,40 +363,95 @@ Imperator.World impWorld
...
-								countyHoldersCache.Add(holderId);
+							var ck3Governorship = LandedTitles[ck3GovernorshipName];
+							if (Characters.TryGetValue(ck3Governorship.GetHolderId(ck3BookmarkDate), out var governor)) {
+								title.ClearHolderHistory();
+								title.SetHolderId(governor.ID, ck3Governorship.GetDateOfLastHolderChange());
 							}
+							title.DeFactoLiege = null;
+							countyHoldersCache.Add(title.GetHolderId(ck3BookmarkDate));
+						} else if (impMonarch is not null) {
+							GiveCountyToMonarch(title, ck3Country, (ulong)impMonarch, ck3BookmarkDate);
```
Comment (Idhrendur) [inline ImperatorToCK3/CK3/World.cs:431]: This is long enough you should consider extracting some functions

File: ImperatorToCK3/CK3/World.cs [hunk 3]
```diff
@@ -314,40 +363,95 @@ Imperator.World impWorld
...
+					Logger.Warn($"Capital barony province not found {title.CapitalBaronyProvince}");
+					continue;
+				}
+				var ck3CapitalBaronyProvince = Provinces[title.CapitalBaronyProvince];
+				var impProvince = ck3CapitalBaronyProvince.ImperatorProvince;
+
+				if (impProvince is not null) {
+					var impCountry = impProvince.OwnerCountry.Value;
+
+					if (impCountry is not null && impCountry.CountryType != CountryType.rebels) {
```
Comment (Idhrendur) [inline ImperatorToCK3/CK3/World.cs:387]: Again, reverse the condition

**Review by Idhrendur — APPROVED** [review]

File: ImperatorToCK3/CK3/World.cs (hunk 2 above)
Comment (IhateTrains) [reply ImperatorToCK3/CK3/World.cs]: Extracted `GiverCountryToGovernor`.

**Pull request merged**